### PR TITLE
Low-risk refactors to cut down on warnings

### PR DESCRIPTION
**Pull request by henricook**
- Added 'silencer' (@silent) to suppress warnings we can't do anything about (because of HMRC libraries or structural changes that we're not going to make just yet, mainly use of Play.application)
- Removed unused vars, default arguments and methods, added flags to the compiler to tell us about them as warnings in future
- Moved Question/Answer out of the PersistedObjects mega file (as it's deprecated and triggers a warning) to become QuestionnaireQuestion and QuestionnaireAnswer

I've run acceptance but of course encourage you to do the same kind reviewer :-)

### CODE REVIEW

File: app/repositories/ReactiveRepositoryHelpers.scala [hunk 1]
```diff
@@ -62,7 +62,7 @@ trait ReactiveRepositoryHelpers {
     }
   }
 
-  private[this] def singleUpdateValidatorImpl(id: String, actionDesc: String, ignoreNotFound: Boolean = false,
+  private[this] def singleUpdateValidatorImpl(id: String, actionDesc: String, ignoreNotFound: Boolean,
```
Comment (benjstephenson) [inline app/repositories/ReactiveRepositoryHelpers.scala]: Why drop the default value?

File: app/services/testdata/InProgressQuestionnaireStatusGenerator.scala [hunk 2]
```diff
@@ -42,61 +42,61 @@ trait InProgressQuestionnaireStatusGenerator extends ConstructiveGenerator {
...
         None
       }
     }
 
     def getHaveDegreeAnswer = {
       if (generatorConfig.isCivilServant) {
-        Some(PersistedQuestion("Do you have a degree?",
-          PersistedAnswer(Some(if (generatorConfig.hasDegree) { "Yes" } else { "No" }), None, None))
+        Some(QuestionnaireQuestion("Do you have a degree?",
+          QuestionnaireAnswer(Some(if (generatorConfig.hasDegree) { "Yes" } else { "No" }), None, None))
```
Comment (benjstephenson) [inline app/services/testdata/InProgressQuestionnaireStatusGenerator.scala]: Definitely a question for the wider team, but - why are we not using booleans here?

File: app/repositories/ReactiveRepositoryHelpers.scala (hunk 1 above)
Comment (henricook) [inline app/repositories/ReactiveRepositoryHelpers.scala]: The new compiler flags said it was unused (the public methods that call this private method all have their own default  values that hide this one)

File: app/services/testdata/InProgressQuestionnaireStatusGenerator.scala (hunk 2 above)
Comment (henricook) [inline app/services/testdata/InProgressQuestionnaireStatusGenerator.scala]: OH TELL ME ABOUT IT - the yes'es and no's get persisted to the database as well, legacy issues. It's possible we can change it for this year, if we talk to the boss and he approves of the work. Of course, change is risky and what's currently there works

File: app/repositories/ReactiveRepositoryHelpers.scala (hunk 1 above)
Comment (benjstephenson) [inline app/repositories/ReactiveRepositoryHelpers.scala]: :+1:

File: app/services/testdata/InProgressQuestionnaireStatusGenerator.scala (hunk 2 above)
Comment (benjstephenson) [inline app/services/testdata/InProgressQuestionnaireStatusGenerator.scala]: I remember having similar aberrations around they way the diversity questions are stored.   Let's chat about it with Fearless Leader.

**Pull request merged**